apache/submarine
# SUBMARINE-831. Update the status of submarine custom resource in syncHandler

### What is this PR for?
Update the status of submarine custom resource in syncHandler so that we can see the status of the submarine custom resource.

Reference: https://github.com/kubernetes/sample-controller/blob/73e81dab82c945b087f7db1f1f4a7b50e7d92751/controller.go#L321-L333

### What type of PR is it?
[Feature]

### Todos

### What is the Jira issue?
https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-831

### How should this be tested?

After running the operator and create the submarine custom resource, try to change the replicas of server and database in the spec. (See the following video)

### Screenshots (if appropriate)

https://user-images.githubusercontent.com/47914085/124008646-3a280d00-da0f-11eb-8cbe-e4e0634716ae.mp4

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No
